pkg/observ/observ.go

```diff
@@ -17,16 +18,62 @@ type observabilityContext struct {
 	spanCtx context.Context
 }
 
-// RegisterTraceExporter returns a jaeger exporter for exporting traces to opencensus.
+// jaegerTraceExporter is used to hold the trace exporter for Jaeger
+type jaegerTraceExporter struct {
+	exporter (*(jaeger.Exporter))
+}
+
+// datadogTraceExporter is used to hold the trace exporter for Datadog
+type datadogTraceExporter struct {
+	exporter (*(datadog.Exporter))
+}
+
+// Exporter might be a JaejerExporter, DataDogExporter, etc
+// All Exporters *must* implement these two functions.
```

Comment:
no need for this line of comment. The compiler enforces interface implementations.